### DISCUSSION TITLE
Per-Instance Block Handler

### DISCUSSION THREAD

**Discussion by HSGamer** [Minestom/Minestom#1023] [Help] [answered]
I'm making a game server that creates an instance for each arena.
I use some ChunkLoader to load the instance from file.
How can I handle the block handler differently for each instance?
Like a TNT block can explode on one instance but shoot fireworks on another one. Or a chest can open its inventory in one instance but drop all items in another one when interacting with it.

**Comment by MrGazdag** [comment, marked as answer]
Well, if you're using a custom ChunkLoader already, then i would suggest modifying it to use a different BlockHandler depending on what kind of instance is being loaded.

**Reply by HSGamer** [reply]
Not really a custom ChunkLoader.
I use the default AnvilLoader and the SlimeLoader from the extension.
In this case, I guess I have to make a decorative ChunkLoader and add the BlockHandler after the original ChunkLoader loads the chunk.
